### PR TITLE
Added validation on parental_consent only for underage.

### LINKED ISSUE #263: Add parental consent custom validation
**Related component**: `client` app, `api` app

## Description
Currently the parental consent field is required. It shouldn't be.

Add a custom validation function that only fires a validation error if the fields are blank if the volunteer is under 18 y/o.

Test this through the `/api/registration` endpoint.

----
### Prerequisites
_[Make sure you've done the following before posting your issue:]_
* [x] I have filled in the correct labels (fill at least one of bug/enhancement)
* [x] I have searched for duplicate issues and found none
* [x] My issue is phrased as a task - it's clear what needs to be done here.

### PR DESCRIPTION
**Pull request by ykrinsky**
Fixes #263

### CODE REVIEW

File: client/validators.py [hunk 1]
```diff
@@ -49,3 +52,20 @@ def digits_of(n):
...
+    """
+    volunteer_age = (date.today() - new_volunteer['date_of_birth']).days / 365
+    if volunteer_age < LEGAL_AGE:
+        try:
+            parental_consent = new_volunteer['parental_consent']
+            parent_id = parental_consent['parent_id']
+            parent_name = parental_consent['parent_name']
+
+        except KeyError:
+            raise ValidationError("Enter parental consent details, as you are underage.")
```
Comment (stavlocker) [inline client/validators.py:71]: Change message to "Volunteers aged 16-18 must enter parental consent details"

File: client/validators.py [hunk 2]
```diff
@@ -49,3 +52,20 @@ def digits_of(n):
...
         raise ValidationError('Invalid ID number: Incorrect check digit')
+
+
+def parental_consent_validator(new_volunteer):
+    """
+    If volunteer age is under LEGAL_AGE, check that received parental_consent details.
+    :raise: ValidationError if volunteer is underage and given data is missing
+    parental_consent or any parental_consent details.
+    """
+    volunteer_age = (date.today() - new_volunteer['date_of_birth']).days / 365
```
Comment (stavlocker) [inline client/validators.py:63]: This doesn't take leap years into account - a person born on 19/4/2002 will be counted as 18y/o in 15/4/2020.

Instead you can use:

```
today = date.today()
if new_volunteer['date_of_birth'] <= today.replace(year=today.year-18):  # underage
```

Comment (stavlocker) [inline client/validators.py:63]: Additionaly, please validate that the user is above 16 y/o and deny the registration if they aren't

File: api/serializers.py [hunk 3]
```diff
@@ -20,6 +27,11 @@ class Meta:
                   'week_assignments_capacity', 'wanted_assignments', 'phone_number', 'email', 'parental_consent',
                   'languages']
 
+    def validate(self, data):
+        parental_consent_validator(data)
```
Comment (stavlocker) [inline api/serializers.py:31]: Does DRF capture the ValidationError thrown here and presents the error message in the JSON? Or does the exception actually raise? (asking to validate)

Comment (ykrinsky) [reply api/serializers.py:31]: DRF catches the exception and returns the message in non_field_error json field as a response.
image

Comment (stavlocker) [reply api/serializers.py:31]: Nice.

File: client/validators.py (hunk 2 above)
Comment (ykrinsky) [reply client/validators.py:63]: Opened #282

**Pull request merged**